code/game/objects/structures/barricade.dm

```diff
@@ -480,25 +484,28 @@
 			balloon_alert(user, "You need at least 2 metal")
 			return FALSE
 
-		balloon_alert_to_viewers("repairing base...")
+		if(LAZYACCESS(user.do_actions, src))
+			return
+
+		balloon_alert_to_viewers("Repairing base...")
 
-		if(!do_after(user, 2 SECONDS, TRUE, src, BUSY_ICON_FRIENDLY) || obj_integrity >= max_integrity)
+		if(!do_after(user, 2 SECONDS, TRUE, src, BUSY_ICON_FRIENDLY))
```

Comment:
why remove the integ check? If someone else is repairing it while you're working on it, you'll waste metal